core/src/types.ts

```diff
@@ -87,39 +95,58 @@ export type TBuilder<OriginalModel> = {
     OriginalModel[K]
   >;
 } & {
-  build<TransformedModel>(
+  build<TransformedModel = OriginalModel>(
     args?: TFieldBuilderArgs<OriginalModel>
   ): TransformedModel;
-  buildGraphql<TransformedModel>(
+  buildGraphql<TransformedModel = OriginalModel>(
```

Comment:
This allows for the function to return the expected type by default which works very handy with the new patterns.